yarn-project/archiver/src/archiver/kv_archiver_store/block_store.ts

```diff
@@ -340,7 +575,14 @@ export class BlockStore {
       txEffects.push(deserializeIndexedTxEffect(txEffect).data);
     }
     const body = new Body(txEffects);
-    const block = new L2Block(archive, header, body, Fr.fromBuffer(blockHash));
+    const block = new L2BlockNew(
+      archive,
+      header,
+      body,
+      CheckpointNumber(blockStorage.checkpointNumber!),
+      0,
```

Comment:
Heads up this should be the proper index within checkpoint

Reply:
Ah, yes good catch!